src/PTY.ts

```diff
@@ -1,32 +1,48 @@
 import * as ChildProcess from "child_process";
 import * as OS from "os";
 import * as _ from "lodash";
-import * as pty from "pty.js";
+import * as pty from "node-pty";
 import {loginShell} from "./utils/Shell";
 import {debug} from "./utils/Common";
 
+let smth = true ? null : pty.createTerminal(undefined, undefined, undefined);
+type ITerminal = typeof smth;
+
 export class PTY {
-    private terminal: pty.Terminal;
+    private terminal: ITerminal;
 
     // TODO: write proper signatures.
     // TODO: use generators.
     // TODO: terminate. https://github.com/atom/atom/blob/v1.0.15/src/task.coffee#L151
     constructor(words: EscapedShellWord[], env: ProcessEnvironment, dimensions: Dimensions, dataHandler: (d: string) => void, exitHandler: (c: number) => void) {
-        const shellArguments = [...loginShell.noConfigSwitches, "-i", "-c", words.join(" ")];
-
+        console.log('constructor');
+        const shellArguments = [...loginShell.noConfigSwitches, "/c", words.join(" ")];
+        console.log(`PTY: ${loginShell.executableName} ${JSON.stringify(shellArguments)}`);
         debug(`PTY: ${loginShell.executableName} ${JSON.stringify(shellArguments)}`);
 
+        console.log('PTY: preparing to fork');
+        console.log(JSON.stringify({
+            cols: dimensions.columns,
+            rows: dimensions.rows,
+            cwd: env.PWD
+        }));
         this.terminal = pty.fork(loginShell.executableName, shellArguments, {
             cols: dimensions.columns,
             rows: dimensions.rows,
             cwd: env.PWD,
             env: env,
         });
+        console.log('PTY: 1');
 
-        this.terminal.on("data", (data: string) => dataHandler(data));
-        this.terminal.on("exit", (code: number) => {
+        (this.terminal as any).on("data", (data: string) => {
+            console.log('PTY: data');
+            dataHandler(data);
+        });
+        (this.terminal as any).on("exit", (code: number) => {
+            console.log('PTY: exit');
             exitHandler(code);
         });
```

Comment:
For some reason, `ITerminal` doesn't declare its `on` method. So I had to do these ugly `as any` casts.